### ISSUE TITLE
Bug in AD57X4R.cpp

### ISSUE THREAD

**Issue by johnmx** [janelia-arduino/AD57X4R#8]
I found a bug in AD57X4R.cpp. In the function _writeMosiDatagramToChip()_, instead of
`Datagram mosi_datagram_n = mosi_datagram_array[chip];`
it should be:
`Datagram mosi_datagram_n = mosi_datagram_array[chip_n];`

**Comment by peterpolidoro** [comment]
Good find! I think I have that bug in several other places too. Probably explains why other people have been experiencing problems too. I have not had a chance to test this code on hardware. I am making a new version 5.0.1 with the bug fixes. Do you have hardware that you can use to confirm whether or not it works properly now? Thanks!

**Comment by johnmx** [comment]
Yes, I tested the library on hardware, and it works. However, I only tested the sawtooth example.

**Comment by peterpolidoro** [comment]
Thank you! Did you test it for one chip or multiple chips?

**Comment by johnmx** [comment]
Only one chip: AD5724R
